=== modified file 'account/account.py'
--- account/account.py	2012-10-09 09:58:14 +0000
+++ account/account.py	2012-11-18 12:56:50 +0000
@@ -1932,6 +1932,17 @@
             ids = self.search(cr, user, args, limit=limit, context=context or {})
         return self.name_get(cr, user, ids, context=context)
 
+
+    def copy(self, cr, uid, id, default={}, context=None, done_list=[], local=False):
+        tax = self.browse(cr, uid, id, context=context)
+        if not default:
+            default = {}
+        default = default.copy()
+        default['description'] = (tax['description'] or '') + '(copy)'
+        default['name'] = (tax['name'] or '') + '(copy)'
+        return super(account_tax, self).copy(cr, uid, id, default, context=context)
+
+
     def write(self, cr, uid, ids, vals, context=None):
         if vals.get('type', False) and vals['type'] in ('none', 'code'):
             vals.update({'amount': 0.0})

=== modified file 'account/account_invoice.py'
--- account/account_invoice.py	2012-10-05 12:58:20 +0000
+++ account/account_invoice.py	2012-11-18 14:50:28 +0000
@@ -1490,7 +1490,7 @@
     #
     # Set the tax field according to the account and the fiscal position
     #
-    def onchange_account_id(self, cr, uid, ids, product_id, partner_id, inv_type, fposition_id, account_id):
+    def onchange_account_id(self, cr, uid, ids, product_id, partner_id, inv_type, fposition_id, account_id, context=None):
         if not account_id:
             return {}
         unique_tax_ids = []
@@ -1501,7 +1501,7 @@
             unique_tax_ids = self.pool.get('account.fiscal.position').map_tax(cr, uid, fpos, taxes)
         else:
             product_change_result = self.product_id_change(cr, uid, ids, product_id, False, type=inv_type,
-                partner_id=partner_id, fposition_id=fposition_id,
+                partner_id=partner_id, fposition_id=fposition_id, context=context,
                 company_id=account.company_id.id)
             if product_change_result and 'value' in product_change_result and 'invoice_line_tax_id' in product_change_result['value']:
                 unique_tax_ids = product_change_result['value']['invoice_line_tax_id']

=== modified file 'account/account_invoice_view.xml'
--- account/account_invoice_view.xml	2012-02-16 18:50:48 +0000
+++ account/account_invoice_view.xml	2012-11-18 14:48:55 +0000
@@ -59,7 +59,7 @@
                         <field name="uos_id" on_change="uos_id_change(product_id, uos_id, quantity, name, parent.type, parent.partner_id, parent.fiscal_position, price_unit, parent.address_invoice_id, parent.currency_id, context, parent.company_id)" nolabel="1"/>
                     </group>
                     <field name="price_unit"/>
-                    <field domain="[('company_id', '=', parent.company_id), ('journal_id', '=', parent.journal_id), ('type', '&lt;&gt;', 'view')]" name="account_id" on_change="onchange_account_id(product_id, parent.partner_id, parent.type, parent.fiscal_position,account_id)"/>
+                    <field domain="[('company_id', '=', parent.company_id), ('journal_id', '=', parent.journal_id), ('type', '&lt;&gt;', 'view')]" name="account_id" on_change="onchange_account_id(product_id, parent.partner_id, parent.type, parent.fiscal_position,account_id, context)"/>
                     <field name="discount" groups="base.group_extended"/>
                     <field domain="[('type','&lt;&gt;','view'), ('company_id', '=', parent.company_id), ('parent_id', '!=', False)]" name="account_analytic_id" groups="analytic.group_analytic_accounting"/>
                     <field name="company_id" groups="base.group_multi_company" readonly="1"/>

